app/controllers/articles_controller.rb

```diff
@@ -87,7 +87,7 @@ def preview
     begin
       fixed_body_markdown = MarkdownFixer.fix_for_preview(params[:article_body])
       parsed = FrontMatterParser::Parser.new(:md).call(fixed_body_markdown)
-      parsed_markdown = MarkdownParser.new(parsed.content, source: Article.new(user: current_user))
+      parsed_markdown = MarkdownParser.new(parsed.content, source: Article.new(user: current_user), user: current_user)
```

Comment:
Here's context in action! The `source` is **where** the liquid tag will be used and `user` is **who** is using it.

Reply:
Is there a case where `source.user` and `user` are different?

Reply:
Currently, no. In the future that could happen. Liquid tags are "used" wherever we render the markdown editor. In the future, it's possible an Admin creates something on behalf of another user.

Also, the relationship isn't always `user` on the `source`. In future source types, it could be `.author`, `.admin`, etc. 

And since these are used in a markdown editor, it's also possible for liquid tags to be used within an object that has no association to a user. `Page` is a current example. In the case of `Page`, we'd just want to check if the currently signed-in user is permitted to use it since `Page` has no `User` association itself.